### PR TITLE
Implemeted Usage cmdlet for Network; Added usage tests

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by EvgenyAgafonchikov** [comment]
@markcowl could you please clarify is this part required now:
> If changes were made to any cmdlet, the XML help was regenerated using the platyPSHelp module.

It looks like there is no XML help file in Network module now. Was something moved/restructured?

**Comment by markcowl** [comment]
@EvgenyAgafonchikov Help is now all in markdown - the xml file is created at runtime.  See: https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md

File: src/ResourceManager/Network/Commands.Network/help/Get-AzureRmNetworkUsage.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+external help file: Microsoft.Azure.Commands.Network.dll-Help.xml
+online version: 
+schema: 2.0.0
+---
+
+# Get-AzureRmNetworkUsage
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/help/Get-AzureRmNetworkUsage.md]: You have to actually fill these in with proper values

File: src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.Automation
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmNetworkUsage")]
+    public partial class GetAzureRmNetworkUsage : NetworkBaseCmdlet
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs]: Need an OutputType attribute, with the proper OutputType

**Comment by markcowl** [comment]
@deepakrajendranmsft can you please review/approve?

File: src/ResourceManager/Network/Commands.Network/help/Get-AzureRmNetworkUsage.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+The Get-AzureRmNetworkUsage cmdlet gets limits and current usage for Network resources.
+
+## EXAMPLES
+
+### Example 1
+```
+PS C:\> Get-AzureRmNetworkUsage -Location westcentralus
+```
+
+Gets resources usage data in westcentralus region
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/help/Get-AzureRmNetworkUsage.md]: Please show a representation of the output of the cmdlet

File: src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.Automation
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmNetworkUsage"), OutputType(typeof(PSUsage))]
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs]: What does the output of this cmdlet look like on screen?  Please see the help comments below and include this.  You will want to make sure that the most relevant information is displayed, and extraneous information is not.

**Comment by DeepakRajendranMsft** [comment]
@markcowl @EvgenyAgafonchikov i have approved the changes

File: src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs (hunk 4 above)
Comment (EvgenyAgafonchikov) [inline src/ResourceManager/Network/Commands.Network/Usage/GetAzureNetworkUsageCommand.cs]: @markcowl, I've updated documentation. Could you please have a look?

File: src/ResourceManager/Network/Commands.Network/Microsoft.Azure.Commands.Network.format.ps1xml [hunk 5]
```diff
@@ -2,6 +2,32 @@
 <Configuration>
   <ViewDefinitions>
     <View>
+      <Name>Microsoft.Azure.Commands.Network.Models.PSUsage</Name>
+      <ViewSelectedBy>
+        <TypeName>Microsoft.Azure.Commands.Network.Models.PSUsage</TypeName>
+      </ViewSelectedBy>
+      <ListControl>
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/Microsoft.Azure.Commands.Network.format.ps1xml]: The ddefault here should be a table view - with only 3 columns, this makes the display much easier.  Users can always use the list view by piping through ```fl```

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1643/

**Pull request merged**